libvast/vast/logger.hpp

```diff
@@ -48,34 +50,93 @@
 
 #define VAST_LOG(...) VAST_PP_OVERLOAD(VAST_LOG_, __VA_ARGS__)(__VA_ARGS__)
 
-#if CAF_LOG_LEVEL >= CAF_LOG_LEVEL_TRACE
+namespace vast::detail {
+template <class T>
+auto id_or_name(T&& x) {
+  static_assert(
+    !std::is_same_v<const char*, std::remove_reference<T>>,
+    "const char* is not allowed for the first argument in a logging statement"
+    "supply a component or use `VAST_[ERROR|WARNING|INFO|DEBUG]_ANON` instead");
```

Comment:
Is `_ANON` now simply a `_` suffix?

Reply:
Yes, I missed updating that error message.
The suffix transitioned from `_FF` to `_ANON` to `_NS` (for *no subject*). But all of them suggest a wrong meaning at some use-sites. Maybe `_CS` for *custom subject*?

Reply:
As per our discussion, reverted to `_ANON`.